app/presenters/guide_presenter.rb

```diff
@@ -41,6 +41,28 @@ def parts
     end
   end
 
+  def hide_chapter_navigation?
+    hide_navigation_elements = content_item.parsed_content["details"]["hide_chapter_navigation"]
+    step_navs = content_item.parsed_content["links"]["part_of_step_navs"]
+
+    part_of_step_navs = false
+    part_of_step_navs = step_navs.length if step_navs
+
+    true if part_of_step_navs && hide_navigation_elements
```

Comment:
You don't need to do `true if`. You can just do:

`part_of_step_navs && hide_navigation_elements`

Ruby will return the result of the last statement regardless.

The same is true for `show_guide_navigation?`.